### PR TITLE
discount system

### PR DESCRIPTION
**Pull request by NourAdawi**
- created two endpoints: one for the admin (POST /promocode) , and one for the customer (POST /redeem)
- admin endpoint will generate a new promo code if it's not already used before and store it in database.
- customer endpoint will check if the code valid/expired/ has been used by the user or not, and then redeem the code.

- Muhammed and I have modified the database and "create data" file a little, we add another model "discountUsers" in order to be able to check if the user used the code before or not.

### CODE REVIEW

File: server/src/controllers/redeemCode.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      );
+
+    const { uid } = req.user;
+    const promocodeId = code.id;
+
+    const data = await UserDiscounts.findOrCreate({
+      where: { promocode: promocodeId, userid: uid },
+    });
+
+    if (!data[1])
```
Comment (Amoodaa) [inline server/src/controllers/redeemCode.js:26]: data[1] or data[0]?

File: server/src/routes/redeemRouter.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const router = require('express').Router();
+const { redeemCode } = require('../controllers');
+
+router.post('/', redeemCode);
+
+module.exports = router;
```
Comment (Amoodaa) [inline server/src/routes/redeemRouter.js:6]: why such small routers, combine the 2 into one `/promo` router, then do POST `/redeem` and POST `/`

**Review by Amoodaa — CHANGES_REQUESTED** [review]
Looks good otherwise

File: server/src/controllers/redeemCode.js (hunk 1 above)
Comment (NourAdawi) [reply server/src/controllers/redeemCode.js:26]: data[0] contains the information, the second index indicates whether the new id has been added by ( findOrCreate ) method or already exits.

File: server/src/routes/redeemRouter.js (hunk 2 above)
Comment (NourAdawi) [reply server/src/routes/redeemRouter.js]: done

**Review by Ahmad-Omar — APPROVED** [review]

File: server/src/routes/index.js [hunk 3]
```diff
@@ -2,13 +2,12 @@ const router = require('express').Router();
 const mealsRouter = require('./meals');
 const categoryRouter = require('./categories');
 const restaurantRouter = require('./restaurants');
-
+const promoCodeRouter = require('./promocode');
 const { auth } = require('../controllers');
 const { decodeToken } = require('../middleware/privateRoute');
```
Comment (Amoodaa) [inline server/src/routes/index.js:7]: can we call it isAuth

**Review by Amoodaa — APPROVED** [review]

**Pull request merged**